internal/jsonfieldnames/extract.go

```diff
@@ -0,0 +1,98 @@
+// Package jsonfieldnames contains a function to extract JSON field names from a value.
+//
+// In addition to their fixed, hardcoded members, business objects for energy marshal additional values into Geschaeftsobjekt.ExtensionData.
+// As Go's standard library JSON package has no option for a kitchen sink field accepting all remaining JSON fields,
+// there are two passes of unmarshaling: Once into the struct, once into a generic map. To avoid duplicates in the struct and the map,
+// fields that can be found in the struct are removed from the map. This package provides a mechanism to extract the names of
+// these fields in a generic way.
+package jsonfieldnames
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// Extract extracts JSON field names from a struct value or a pointer to a struct value. If the value passed to Extract is nil,
+// is not a struct value or a pointer to a struct value, an error is returned instead.
+func Extract(value interface{}) ([]string, error) {
+	typ := reflect.TypeOf(value)
+	if typ == nil {
+		return nil, fmt.Errorf("cannot extract field names: value is nil")
```

Comment:
```suggestion
		return nil, fmt.Errorf("cannot extract field names: typ is nil")
```
zumindest ist das, was in der zeile drüber steht ;) wie sich die reflection verhält, wenn value nil ist, weiß ich nicht.

Reply:
nil-Values sind kein Problem, solange der konkrete Typ zur Laufzeit bekannst ist. Ein Struct-Pointer, der nil ist, ist okay und der Typ kann ermittelt werden, bei einer Variablen, die ein Interface-Typ ist, geht das nicht.